node/primitives/src/currency.rs

```diff
@@ -187,7 +190,8 @@ macro_rules! create_currency_id {
 					Self::LPToken(token_symbol_1, token_type_1, token_symbol_2, token_type_2) => {
 						(((*token_symbol_1 as u64) << 16) & 0x0000_0000_00ff_0000) + (((*token_type_1 as u64) << 24) & 0x0000_0000_ff00_0000) +
 						(((*token_symbol_2 as u64) << 32) & 0x0000_00ff_0000_0000) + (((*token_type_2 as u64) << 40) & 0x0000_ff00_0000_0000) + discr
-					}
+					},
+					_ => 0 as u64,
```

Comment:
0 represents CurrencyId::Native(TokenSymbol::ASG). We should not provide a wildcard here as each one of bifrost currencies needs to have a u64 presentation.

Since the discriminant for Erc20 is 7, we can compose an Erc20 currencyId like (7 <<8 + x), while x means tokensymbol related to the Erc20 token.

Reply:
x needs not to be the lowest 8 bits. Instead, it can make use of the upper 48 bits, as long as it keeps  the 7 discriminant in the right place.